### PR TITLE
False positive in SensitiveDataHeuristics - exclude certification from maybeCertificate() regex

### PR DESCRIPTION
**Pull request by ockers**
# Description

Exclude the term certification from maybeCertificate regex.  This has caused false positive warnings for our application, which does OS snapshot certifications, about logging sensitive data (certificate) in exception logs.

## Testing Done

Tested using python re:

```
#!/usr/bin/python3

import re

# this regex is from https://github.com/github/codeql/blob/ec0066d/python/ql/lib/semmle/python/security/internal/SensitiveDataHeuristics.qll#L78
# string maybeCertificate() { result = "(?is).*(cert)(?!.*(format|name)).*" }

test_strings = ["nothing", "certificate", "certificate name", "certificate format", "certname", "certification"]

for teststring in test_strings:

    oldpattern = "(?is).*(cert)(?!.*(format|name)).*"
    newpattern = "(?is).*(cert)(?!.*(format|name|ification)).*"

    oldresult = re.match(oldpattern, teststring)
    newresult = re.match(newpattern, teststring)

    print(f"Result using old regex on {teststring} is {oldresult}")
    print(f"Result using new regex on {teststring} is {newresult}\n")
```

Output shows that the string "certification" doesn't match the new regex:

```
$ ./certification.py 
Result using old regex on nothing is None
Result using new regex on nothing is None

Result using old regex on certificate is <re.Match object; span=(0, 11), match='certificate'>
Result using new regex on certificate is <re.Match object; span=(0, 11), match='certificate'>

Result using old regex on certificate name is None
Result using new regex on certificate name is None

Result using old regex on certificate format is None
Result using new regex on certificate format is None

Result using old regex on certname is None
Result using new regex on certname is None

Result using old regex on certification is <re.Match object; span=(0, 13), match='certification'>
Result using new regex on certification is None
```

### CODE REVIEW

**Review by geoffw0 — COMMENTED** [review]
Seems reasonable, and I appreciate that you have tested the regex.  :+1:

I'm not sure if the change note is in the right place / format.  I'll let someone else check that and (if it is) they can approve the PR.

**Comment by jketema** [comment]
For change notes please follow the instructions provided here: https://github.com/github/codeql/blob/main/docs/change-notes.md

File: javascript/ql/lib/change-notes/2024-01-30-certification_not_certificate.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* The string "certification" isn't a certificate, and should be excluded from the maybeCertificate() regex.
```
Comment (erik-krogh) [inline javascript/ql/lib/change-notes/2024-01-30-certification_not_certificate.md:4]: Maybe focus more on what the visible change is to end users?

Maybe something like: 

```
The name `certification` is no longer seen as possibly being a certificate, and will therefore no longer be flagged in queries like `clear-text-logging` that look for sensitive data.
```

Comment (ockers) [reply javascript/ql/lib/change-notes/2024-01-30-certification_not_certificate.md]: Updated to be more nearly like that text, please review at your convenience.

**Review by erik-krogh — APPROVED** [review]

**Review by hmac — APPROVED** [review]

**Pull request merged**